lib/phlex.rb

```diff
@@ -7,6 +7,9 @@
 loader.setup
 
 module Phlex
+  Error = Module.new
+  ArgumentError = Class.new(ArgumentError) { include Error }
```

Comment:
This trick allows us to have a `Phlex::ArgumentError` that `is_a?(ArgumentError)` and `is_a?(Phlex::Error)`.